docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,684 @@
+[INFO]
+[INFO] Greeting Extension - Parent                                        [pom]
+[INFO] Greeting Extension - Runtime                                       [jar]
+[INFO] Greeting Extension - Deployment                                    [jar]
+[INFO]
+...
+[INFO]
+-------------------------------------------------------
+ T E S T S
+-------------------------------------------------------
+Running org.acme.quarkus.greeting.deployment.GreetingTest
+Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec
+
+Results :
+
+Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
+
+[INFO] ------------------------------------------------------------------------
+[INFO] Reactor Summary for Greeting Extension - Parent 1.0-SNAPSHOT:
+[INFO]
+[INFO] Greeting Extension - Parent ........................ SUCCESS [  0.081 s]
+[INFO] Greeting Extension - Runtime ....................... SUCCESS [  1.044 s]
+[INFO] Greeting Extension - Deployment .................... SUCCESS [  1.253 s]
+[INFO] ------------------------------------------------------------------------
+[INFO] BUILD SUCCESS
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time:  2.479 s
+[INFO] Finished at: 2019-07-03T12:00:00+01:00
+[INFO] ------------------------------------------------------------------------
+-------------
+
+Looks good!! Congratulations you just finished your first extension.
+
+=== Debugging your extension
+
+_If debugging is the process of removing bugs, then programming must be the process of putting them in._
+Edsger W. Dijkstra
+
+
+*Debugging your application build*
+
+Since your extension deployment is made during the application build, this process is triggered by your build tool.
+That means if your want to debug this phase you need to launch your build tool with the remote debug mode switched one.
+
+_Maven_
+
+You can activate Maven remote debugging by using `mvnDebug`. You can launch your application with the following command line:
+
+`mvnDebug clean compile quarkus:dev`
+
+By default, Maven will wait for a connection on `localhost:8000`. Now, you can run your IDE `Remote` configuration to attach it to `localhost:5000`.
+
+_Gradle_
+
+TODO
+
+*Debugging your test extension*
+
+We have seen together how to test your extension and sometimes things don't go so well and you want to debug your tests.
+Same principle here, the trick is to enable the Maven Surfire remote debugging in order to attach an IDE `Remote` configuration.
+
+[source, shell]
+-------------
+$ cd `./greeting`
+$ mvn clean test -Dmaven.surefire.debug
+-------------
+
+By default, Maven will wait for a connection on `localhost:5005`.
+
+=== Extension publication
+
+Now that you just finish to build your first extension you should be eager to use it in a Quarkus application! 
+
+*Classic Maven publication*
+
+Because your extension produces a classic JAR, the easiest way to share your extension is to publish it to a Maven repository.
```

Comment:
Given that it produces more than one artifact, let's probably use the plural form, i.e. `...produces classic JARs...`